pkg/operator/controller/externaldns/controller.go

```diff
@@ -172,10 +176,30 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	_, currentDeployment, err := r.ensureExternalDNSDeployment(ctx, r.config.Namespace, r.config.Image, sa, externalDNS)
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to ensure externalDNS deployment: %w", err)
+		res := reconcile.Result{}
+		if apierrors.IsNotFound(err) {
+			if status := apierrors.APIStatus(nil); errors.As(err, &status) {
+				details := status.Status().Details
+				if details != nil {
+					// credentials secret or trusted CA configmap was not synced yet or doesn't exist at all,
+					// either way: no need to requeue immediately polluting the logs.
+					switch details.Kind {
+					case "secrets":
+						res.RequeueAfter = r.config.RequeuePeriod
+						// show that the secret is not there yet
+						if err := r.updateExternalDNSStatus(ctx, externalDNS, currentDeployment, false); err != nil {
+							reqLogger.Error(err, "failed to update externalDNS custom resource")
+						}
+					case "configmaps":
+						res.RequeueAfter = r.config.RequeuePeriod
+					}
+				}
+			}
+		}
```

Comment:
Not against it, but there seems to be 3 sources to throw `NotFound` error (`getSecret()`, `getConfigMap()` and `getDeployment()`), we could have just relied on that if `ensureExternalDNSDeployment()` only ensured deployment.

Looks like `ensureExternalDNSDeployment()` tries to ensure its prerequisites as well, maybe these should be outside at the controller itself.

Reply:
Yeah, that was my thought too, but trying to make a smallest change possible got me :)
Let me try to check the pre-requisites before ensuring the deployment.